docs/okeps/okep-5375-trust-zones.md

```diff
@@ -0,0 +1,265 @@
+
+Security is the primary driver for this feature, while scalability
+considerations are a secondary benefit.
+
+## Goals
+
+Allow cluster admins to dedicate chassis to particular trust zones, only building
+Geneve tunnels between directly connected zones.
+
+Make sure that a rogue chassis cannot force other chassis to create unintended
+tunnels to the rogue chassis.
+
+## Non-Goals
+
+Automatically determining zone assignment for chassis based on workload
+attributes (e.g. UDN attachments).
+
+Implementation of the feature for Central or multi-node IC modes. (See detailed
+rationale below in the `Risks, Known Limitations and Mitigations` section.)
+
+## Introduction
+
+Inter-chassis tunnels carry some cost, in terms of security, performance, and
+scale. Even when a tunnel is not used, chassis attempt to create corresponding
+Interfaces.
+
+This is unnecessary and may even expose victim workloads to an attacker if they
+manage to hijack a chassis, since the attacker then has direct tunnels to all
+chassis.
+
+This OKEP exposes trust zone functionality through a new `TrustZone` CRD. When
+created, a `TrustZone` CR will affect which chassis have tunnels between them,
+and which don't.
+
+Only cluster admin should be able to create `TrustZone` CRs.
+
+## User-Stories/Use-Cases
+
+### Story 1: Isolate a set of chassis in a trust zone
+
+As a cluster admin, I want to assign a set of chassis to a trust zone, so that
+only a chassis that belongs to the trust zone creates tunnel endpoints to other
+chassis from the zone.
+
+### Story 2: Group a set of chassis in multiple trust zones
+
+As a cluster admin, I want to apply unique trust zones to disjoint subsets of
+chassis, so that each subset can tunnel traffic internally but not between
+zones.
+
+## Proposed Solution
+
+### At a Glance
+
+OVN-Kubernetes will maintain the list of remote `Chassis` in SBDB so that only
+chassis that share a trust zone with the local chassis will have a
+corresponding `Chassis` record in SBDB. Since there won't be `Chassis` records
+for chassis from disjoint trust zones, no tunnels will be created.
+
+To guarantee that a rogue chassis won't be able to change trust zone membership
+for itself or other chassis, changing `TrustZone` CRs will not be denied using
+RBAC unless the user is a cluster admin.
+
+### API Details
+
+#### TrustZone CRD
+
+A new `TrustZone` CRD will be created in `k8s.ovn.org/v1alpha1` API group as
+follows:
+
+```
+apiVersion: k8s.ovn.org/v1alpha1
+kind: TrustZone
+metadata:
+  name: trust-zone-a
+spec:
+  nodeSelector:
+    matchLabels:
+      tenant: tenant-a
+status:
+  members:
+    - ovnkube-pod-1
+    - ovnkube-pod-2
+    - ovnkube-pod-3
+  conditions:
+    - type: Ready
+      status: "True"
+      reason: AllTunnelsEstablished
+      lastTransitionTime: "2025-08-13T15:04:05Z"
+```
+
+In the above example, the `TrustZone` named `trust-zone-a` selects nodes that
+are assigned to a tenant using a `tenant` label. The `status.members` field
```

Comment:
what if the rogue user labels node-X to be part of `tenant-a`?

an attacker escapes out of the VM/Pod and is on the node and uses the `kubelet` kube configuration to label the node to be part of the different trust zone

Reply:
I will double check but I assume `./go-controller/pkg/ovnwebhook/nodeadmission.go` should make sure that non-whitelisted annotations won't be allowed for modification by a rogue ovnkube-node.

Reply:
selecting on annotation is not a common kubernetes practice. I think it would make sense to just add label checks in the nodeadmission webhook we already use today.